bundles/org.openhab.binding.mqtt/src/main/java/org/openhab/binding/mqtt/internal/MqttBrokerHandlerFactory.java

```diff
@@ -125,4 +125,13 @@ public void unsubscribe(MQTTTopicDiscoveryParticipant listener) {
             handlers.forEach(broker -> broker.unregisterDiscoveryListener(listener, topic));
         });
     }
+
+     @Override
+     public void publish(String topic, byte[] payload) {
+         handlers.forEach(handler -> {
+             handler.getConnectionAsync().thenAccept(connection -> {
```

Comment:
what if a broker is added later on? Shouldn't that one get a message too? The `MQTTTopicDiscoveryService .subscribe` method is handled this way should this one not also be handled that way?

Reply:
I didn't consider it, but now, thinking at the logic, If I subscribe as a discovery service I am expecting to receive from brokers that were later added since I don't interact with a broker, I interact with this abstraction layer on top of them. But subscription is something continuous in time.

On the other hand, publishing is instantaneous and happens when the user presses the scan button (or when the binding decides to do an active scan), I consider reasonable the expectation to publish only on currently connected brokers, not publish in the future when a new broker is registered

Reply:
If there is no background scan involved and scanning only happens when manually triggered, I think it is ok to leave it as-is. It's an absolute corner-case that a broker is added after the scan was started but before it is finished (especially when the user has to trigger the scan).

@aisebastian, can you confirm that this is the case?

Reply:
To the best of my knowledge, scanning is only done on demand by the user through ... "Thing search", and in terms of when this "scan publish" occurs, it is entirely up to each binding to use it or not